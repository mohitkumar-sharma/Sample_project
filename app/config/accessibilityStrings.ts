export default {
  NORMAL_BUTTON_TEXT: 'normal-button-text',
  SLIDER_BUTTON_TEXT: 'slider-button-text',
  NORMAL_BUTTON: 'normal-button',
  SLIDER_BUTTON_ICON: 'slider-button-icon',
  SLIDER_BUTTON_CONTAINER: 'slider-button-container',
  BANNER_TEXT: 'banner-text',
  HEADER_SIDES_TEXT: 'header-sides-text',
  HEADER_PRESSABLE_ELEMENT: 'header-pressable-element',
  SETTINGS_SWITCH_TEXT: 'settings-switch-text',
  SETTINGS_SWITCH: 'settings-switch',
  INV_TEXT_INPUT: 'inv-text-input',
};
