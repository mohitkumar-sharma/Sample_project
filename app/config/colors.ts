export default {
  black: '#000000',
  gray: '#9B9B9B',
  divider: '#E8E8E8',
  white: '#FFFFFF',
  themeBlue: '#6EB1F7',
  themeLightBlack: '#34424A',
  appTheme: '#1A1924',
  borderLine: '#59606A',
  warning: '#FFCC00',
  red: '#FF0000',
};
