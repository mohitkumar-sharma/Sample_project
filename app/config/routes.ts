export default {
  WELCOME: 'Welcome',
  HOME: 'Home',
  SETTINGS: 'Settings',
};
