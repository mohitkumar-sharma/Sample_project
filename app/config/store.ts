import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit';
import { rootReducer } from '../states';

const store = configureStore({
  reducer: rootReducer,
  middleware: getDefaultMiddleware(),
});

export type AppDispatch = typeof store.dispatch;

export { store };
