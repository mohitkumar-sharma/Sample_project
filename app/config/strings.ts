export default {
  GO_TO_HOME: 'Go To Home',
  DIAMOND_ICON: 'diamond',
  SLIDE_ME: 'Slide me to continue',
  APP_RUNNING_WARNING: 'Notice: You are running this app on',
  WELCOME_INPUT_PLACEHOLDER: 'Enter your name here...',
  SIGN_OUT: 'SignOut',
  HOME_BUTTONS_HEADING: '4 variations of a button',
  PRESS_ME: 'Press me',
  TRANSPARENT_BTN_PRESSED: 'Transparent button is pressed',
  LIGHT_BLACK_BTN_PRESSED: 'Light black button is pressed',
  BLUE_BTN_PRESSED: 'Blue button is pressed',
  SWITCH_TEXT: 'Enable Warning Banner',
  ALERT_TITLE_INVALID_USER_NAME: 'Invalid UserName!',
  ALERT_TITLE_BTN_PRESSED: 'Button Pressed!',
  ALERT_ENTER_VALID_USER_NAME: 'Please enter a valid user name.',
};
