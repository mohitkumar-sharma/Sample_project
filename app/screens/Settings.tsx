import React from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import config from '../config';
import { INVBanner, SettingsSwitchCell } from '../components';
import { DeviceInfo, SettingsInfo } from '../states';
import { useSelector, useDispatch } from 'react-redux';
import { StackNavigationProp } from '@react-navigation/stack';
import { MainStackParamList } from '../navigation/Main';
import { useBackButton } from '../hooks';

/* Props for navigation*/
type NavigationProps = {
  navigation: StackNavigationProp<MainStackParamList, 'Settings'>;
};

const Settings: React.FC<NavigationProps> = ({ navigation }): JSX.Element => {
  /* useDispatch hook from react-redux for dispatching actions*/
  const dispatch = useDispatch();

  /* useSelector hook from react-redux to select device type from redux */
  const deviceType = useSelector(DeviceInfo.selectDeviceType);
  /* useSelector hook from react-redux to select warning banner from redux */
  const isShowBanner = useSelector(SettingsInfo.selectWarningBannerStatus);

  /**
   * This function is responsible to handle the hardware button of android
   */
  const backButtonHandler = () => {
    navigation.goBack();
    return true;
  };

  /* custom hook called from hook directory */
  useBackButton(backButtonHandler);

  /**
   * This function is responsible to dispatch an action to save switch value in to redux
   * @param value : a boolean value
   */
  const onSwitchValueChange = (value: boolean) =>
    dispatch(SettingsInfo.saveWarningBannerStatus({ isShowWarningBanner: value }));

  return (
    <SafeAreaView style={styles.container}>
      {isShowBanner && deviceType !== 'Device' ? (
        <INVBanner text={`${config.strings.APP_RUNNING_WARNING} ${deviceType}`} />
      ) : null}
      <SettingsSwitchCell
        text={config.strings.SWITCH_TEXT}
        onSwitchValueChange={onSwitchValueChange}
        containerStyle={styles.switchContainer}
        switchValue={isShowBanner}
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: { flex: 1, backgroundColor: config.colors.appTheme },
  switchContainer: { paddingVertical: 10, backgroundColor: config.colors.themeBlue },
});

export default Settings;
