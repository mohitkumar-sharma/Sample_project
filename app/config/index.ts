import colors from './colors';
import localConfig from './localConfig';
import strings from './strings';
import routes from './routes';
import fontWeights from './fontWeights';
import accessibilityStrings from './accessibilityStrings';
import { store } from './store';

const config = {
  colors,
  ...localConfig,
  store,
  strings,
  routes,
  fontWeights,
  accessibilityStrings,
};

export default config;
