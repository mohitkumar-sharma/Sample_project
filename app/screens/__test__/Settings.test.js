import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Settings from '../Settings';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import config from '../../config';

describe('Settings', () => {
  const dispatch = jest.fn();
  redux.useDispatch.mockReturnValue(dispatch);

  const navigation = {
    navigate: jest.fn(),
    setOptions: jest.fn(),
  };

  /* Mocking store */
  const mockStore = configureStore([]);

  it('matches snapshot', () => {
    const store = mockStore({ deviceInfo: { deviceType: 'Simulator' }, isShowWarningBanner: true });
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementationOnce(() => store.getState().isShowWarningBanner)
      .mockImplementationOnce(() => store.getState().deviceInfo);
    const { toJSON } = render(<Settings navigation={navigation} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('switch value to be false', () => {
    const { getByTestId } = render(<Settings navigation={navigation} />);
    const switchButton = getByTestId(config.accessibilityStrings.SETTINGS_SWITCH);
    expect(switchButton.props.value).toBe(false);
  });

  it('switch value changes', () => {
    const expectedPayload = [];
    const settingsInfo = { isShowWarningBanner: false };
    const store = mockStore({ settingsInfo });

    jest.spyOn(redux, 'useSelector').mockImplementationOnce(() => store.getState().deviceInfo);

    const { getByTestId } = render(<Settings navigation={navigation} />);
    const switchButton = getByTestId(config.accessibilityStrings.SETTINGS_SWITCH);
    fireEvent(switchButton, 'onValueChange', true);

    store.dispatch({ type: 'SETTINGSINFO', payload: settingsInfo });
    expectedPayload.push({ type: 'SETTINGSINFO', payload: settingsInfo });
    expect(store.getActions()).toEqual(expectedPayload);
  });
});
