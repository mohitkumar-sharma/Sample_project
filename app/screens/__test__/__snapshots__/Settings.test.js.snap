// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#1A1924",
      "flex": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#FFCC00",
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 15,
            "fontWeight": "400",
            "margin": 10,
            "padding": 3,
          },
          undefined,
        ]
      }
      testID="banner-text"
    >
      Notice: You are running this app on true
    </Text>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        },
        Object {
          "backgroundColor": "#6EB1F7",
          "paddingVertical": 10,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 15,
            "marginHorizontal": 15,
            "paddingVertical": 5,
          },
          undefined,
        ]
      }
      testID="settings-switch-text"
    >
      Enable Warning Banner
    </Text>
    <RCTSwitch
      accessibilityRole="switch"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#1A1924"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Array [
            Object {
              "marginRight": 15,
            },
            undefined,
          ],
        ]
      }
      testID="settings-switch"
      tintColor="#E8E8E8"
      value={false}
    />
  </View>
</RCTSafeAreaView>
`;
